src/trio/testing/_raises_group.py

```diff
@@ -303,32 +322,108 @@ class RaisesGroup(ContextManager[ExceptionInfo[BaseExceptionGroup[E]]], SuperCla
 
         def __new__(cls, *args: object, **kwargs: object) -> RaisesGroup[E]: ...
 
+    # allow_unwrapped=True requires: singular exception, exception not being
+    # RaisesGroup instance, match is None, check is None
+    @overload
+    def __init__(
+        self,
+        exception: type[E] | Matcher[E],
+        *,
+        allow_unwrapped: Literal[True],
+        flatten_subgroups: bool = False,
+        match: None = None,
+        check: None = None,
```

Comment:
```suggestion
```

I don't think it's necessary to provide these. a) we're adding the param now so no backcompat issues and b) I don't think we should care about backcompat for types because they're just hints.

Reply:
Hmm, I think it doesn't hurt to have them, and allows users to do some automated stuff where they pass variables that they know to be `None`:
```python
for allow_unwrapped,match in zip([True, None], [False, "hello"]):
  with RaisesGroup(Exception, allow_unwrapped=allow_unwrapped, match=match):
    ...
```